src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -94,17 +106,48 @@ internal void Update(double value, ReadOnlySpan<KeyValuePair<string, object>> ta
 
         internal void SnapShot()
         {
-            var indexSnapShot = Math.Min(this.metricPointIndex, MaxMetricPoints - 1);
-
-            for (int i = 0; i <= indexSnapShot; i++)
+            this.batchSize = 0;
+            for (int i = 0; i < MaxMetricPoints; i++)
```

Comment:
if we start filling from the beginning, then we might not have to navigate the entire MaxMetricPoints of this array.

Reply:
Agree, pretty sure I can optimize this...